### PR TITLE
V14.1: TinyMCE plugins

### PR DESCRIPTION
**Pull request by iOvergaard**
## Description

Update the docs with code examples for V14 on how to add free and premium plugins to the rich text editor.

## Type of suggestion

* [ ] Typo/grammar fix
* [x] Updated outdated content
* [ ] New content
* [x] Updates related to a new version
* [ ] Other

## Product & version (if relevant)
14.1.0

## Deadline (if relevant)

Most of these examples will only work from 14.1 except for the "write your own plugin" section, which already works for 14.0. You can decide if you want to include it now or wait for the 14.1 release in July.

### CODE REVIEW

**Comment by eshanrnh** [comment]
Thanks for the PR, @iOvergaard 🙌 We will look at it as soon as we can.

**Comment by iOvergaard** [comment]
This can be tested on the latest nightly build which has the version `Umbraco.Cms.14.1.0--rc.preview.2.gfe559c2` as of this time writing this.

File: 14/umbraco-cms/fundamentals/backoffice/property-editors/built-in-umbraco-property-editors/rich-text-editor/rte-plugins.md [hunk 1]
```diff
@@ -92,73 +64,139 @@ Here we are loading a custom plugin called `myrteplugin` and adding a button to
 
 <figure><img src="images/my-rte-button-editor.jpg" alt="Rich text editor showing a custom button"><figcaption><p>The text "Hello World!" shows up after clicking the button</p></figcaption></figure>
 
-{% tabs %}
-{% tab title="appsettings.json" %}
+**Add a manifest file**
```
Comment (jonat123) [inline 14/umbraco-cms/fundamentals/backoffice/property-editors/built-in-umbraco-property-editors/rich-text-editor/rte-plugins.md:67]:
```suggestion
1.**Add a manifest file**
```

File: 14/umbraco-cms/fundamentals/backoffice/property-editors/built-in-umbraco-property-editors/rich-text-editor/rte-plugins.md [hunk 2]
```diff
@@ -92,73 +64,139 @@ Here we are loading a custom plugin called `myrteplugin` and adding a button to
...
+        }
+    ]
+}
 ```
 
-{% endtab %}
+{% endcode %}
 
-{% tab title="App_Plugins/MyRtePlugin/plugin.js" %}
+**Add the plugin.js file**
```
Comment (jonat123) [inline 14/umbraco-cms/fundamentals/backoffice/property-editors/built-in-umbraco-property-editors/rich-text-editor/rte-plugins.md:99]:
```suggestion
2.**Add the plugin.js file**
```

File: 14/umbraco-cms/fundamentals/backoffice/property-editors/built-in-umbraco-property-editors/rich-text-editor/rte-plugins.md [hunk 3]
```diff
@@ -92,73 +64,139 @@ Here we are loading a custom plugin called `myrteplugin` and adding a button to
...
+
+## Adding a premium plugin
+
+To add a premium plugin, you need to add the plugin name to the `plugins` array in the `config` object of a `tinyMcePlugin` extension. You also need to add a JavaScript module that can load up the cloud-hosted TinyMCE premium plugins bundle.
+
+{% hint style="info" %}
+Premium plugins require a subscription at [TinyMCE Cloud](https://www.tiny.cloud/). You can go there and sign up for a free trial. You will get a Cloud API key that you can use to try out the premium plugins.
+{% endhint %}
+
+**Declaring the plugin**
```
Comment (jonat123) [inline 14/umbraco-cms/fundamentals/backoffice/property-editors/built-in-umbraco-property-editors/rich-text-editor/rte-plugins.md:154]:
```suggestion
1.**Declaring the plugin**
```

File: 14/umbraco-cms/fundamentals/backoffice/property-editors/built-in-umbraco-property-editors/rich-text-editor/rte-plugins.md [hunk 4]
```diff
@@ -92,73 +64,139 @@ Here we are loading a custom plugin called `myrteplugin` and adding a button to
...
+                }
+            }
+        }
+    ]
+}
+```
+
+{% endcode %}
+
+**Creating the plugin.js file**
```
Comment (jonat123) [inline 14/umbraco-cms/fundamentals/backoffice/property-editors/built-in-umbraco-property-editors/rich-text-editor/rte-plugins.md:184]:
```suggestion
2.**Creating the plugin.js file**
```

**Comment by iOvergaard** [comment]
Thanks for testing and fixing :)

File: 14/umbraco-cms/fundamentals/backoffice/property-editors/built-in-umbraco-property-editors/rich-text-editor/rte-plugins.md [hunk 5]
```diff
@@ -20,65 +16,41 @@ TinyMCE has a lot of open-source plugins available. You can find a list of these
...
-
-The example above shows how to add the open-source [Word Count Plugin](https://www.tiny.cloud/docs/tinymce/6/wordcount/) to the rich text editor. The plugin is added to the `Plugins` array in the configuration. The plugin itself will be shown in the statusbar of the rich text editor, so the `statusbar` option is also added to the `CustomConfig` object.
-
-{% embed url="<https://www.youtube.com/watch?v=BhVeQL0Vq40>" %}
-Rich Text Editor: Adding Plugins
-{% endembed %}
-
-## Adding a premium plugin
+{% hint style="info" %}
+The manifest file should be placed in a folder in `App_Plugins/{YourPackageName}`, with the name `umbraco-package.json`. Read more about [package manifests](../../../../../extending/property-editors/package-manifest.md).
```
Comment (jonat123) [inline 14/umbraco-cms/fundamentals/backoffice/property-editors/built-in-umbraco-property-editors/rich-text-editor/rte-plugins.md:26]:
```suggestion
The manifest file should be placed in a folder in `App_Plugins/{YourPackageName}`, with the name `umbraco-package.json`. Learn how to use the [package manifests](../../../../../extending/property-editors/package-manifest.md).
```

**Review by jonat123 — APPROVED** [review]
@iOvergaard LGTM 😄

I will make sure that it is merged in when 14.1 is released :)

**Pull request merged**